sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,398 @@
+import {
+  ASSESSMENT_DATA_MARRIED,
+  ASSESSMENT_DATA_SINGLE_DEPENDENT,
+  ASSESSMENT_DATA_SINGLE_INDEPENDENT,
+  AWARDS_ONE_OF_TWO_DISBURSEMENT,
+  AWARDS_SINGLE_DISBURSEMENT,
+  AWARDS_SINGLE_DISBURSEMENT_NO_BC_FUNDING,
+  AWARDS_TWO_OF_TWO_DISBURSEMENT,
+  JANE_MONONYMOUS_FROM_OTHER_COUNTRY,
+  JOHN_DOE_FROM_CANADA,
+  OFFERING_FULL_TIME,
+  PROGRAM_GRADUATE_DIPLOMA_WITH_INSTITUTION_PROGRAM_CODE,
+  PROGRAM_UNDERGRADUATE_CERTIFICATE_WITHOUT_INSTITUTION_PROGRAM_CODE,
+  WORKFLOW_DATA_DEPENDANT_RELATIONSHIP_OTHER_LIVING_WITH_PARENTS,
+  WORKFLOW_DATA_MARRIED_WITH_DEPENDENTS,
+  WORKFLOW_DATA_SINGLE_INDEPENDENT_WITH_NO_DEPENDENTS,
+} from "./models/data-inputs";
+import { QueueProcessSummaryResult } from "../../../../../../processors/models/processors.models";
+import {
+  assessmentIdToText,
+  disbursementIdToText,
+  getSuccessSummaryMessages,
+} from "./utils/string-utils";
+import { createIER12SchedulerJobMock, isValidFileTimestamp } from "./utils";
+
+describe(describeProcessorRootTest(QueueNames.IER12Integration), () => {
+  let app: INestApplication;
+  let processor: IER12IntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sharedProgramYearPrefix: number;
+  let referenceSubmissionDate: Date;
+  let getFileNameAsCurrentTimestampMock: jest.SpyInstance;
+  const defaultApplicationNumber = "9900000001";
+
+  beforeAll(async () => {
+    process.env.INSTITUTION_REQUEST_FOLDER = "Institution-Request";
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(IER12IntegrationScheduler);
+    // Default program year prefix.
+    sharedProgramYearPrefix = 2000;
+    referenceSubmissionDate = new Date("2000-06-01");
+    // Intercept file timestamp.
+    getFileNameAsCurrentTimestampMock = jest.spyOn(
+      dateUtils,
+      "getFileNameAsCurrentTimestamp",
+    );
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Update all applications to Overwritten.
+    await db.application.update(
+      { applicationStatus: Not(ApplicationStatus.Overwritten) },
+      { applicationStatus: ApplicationStatus.Overwritten },
+    );
+    // Avoid issues for queries relying on the application number (e.g. hasMultipleApplicationSubmissions).
+    await db.application.update(
+      { applicationNumber: defaultApplicationNumber },
+      { applicationNumber: "0000000000" },
+    );
+  });
+
+  it("Should generate an IER12 file with two records for a single student with no dependents and two disbursements, one sent and one pending.", async () => {
+    // Arrange
+    const testInputData = {
+      student: JOHN_DOE_FROM_CANADA,
+      application: {
+        applicationNumber: defaultApplicationNumber,
+        studentNumber: "12345678",
```

Comment:
for my understanding, I see `studentNumber` in each TC is different, is there a reason?

Reply:
No particular reason, just to produce a different output in the line validation.

Reply:
got it, Just to refersh my mind, if the student doesn't provide a student number in the form, system will append space in place of student number right?

Reply:
Yes, it is treated as a regular text ("Alphanumeric with trailing whitespace"), if not provided it will be all white spaces.
I will change one scenario to have the student number removed, so we will have a numeric, alphanumeric, and not provided one.